src/lib/modules/executor/executor.go

```diff
@@ -0,0 +1,105 @@
+package executor
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/bench-routes/bench-routes/tsdb/file"
+	"github.com/go-ping/ping"
+)
+
+type Ping struct {
+	Min time.Duration `json:"minVal"`
+	Max time.Duration `json:"maxVal"`
+	Avg time.Duration `json:"avgVal"`
+}
+
+type Jitter struct {
+	Min time.Duration `json:"minVal"`
+	Max time.Duration `json:"maxVal"`
+	Avg time.Duration `json:"avgVal"`
+}
+
+type Monitor struct {
+	ResDelay  time.Duration `json:"resDelay"`
+	ResLength int64         `json:"resLength"`
+}
+
+// ExecuteMachine executes ping and jitter operations
+func ExecuteMachine(app file.Appendable, host string) error {
+	pinger, err := ping.NewPinger(host)
+	if err != nil {
+		return fmt.Errorf("error creating ping : %w", err)
+	}
+	pinger.Count = 5
+	var lastTime time.Duration
+	var sum time.Duration
+	var minJitter time.Duration = time.Hour * 24
+	var maxJitter time.Duration = 0
+	// Calculating jitter using ping values
+	pinger.OnRecv = func(pkt *ping.Packet) {
+		if lastTime != time.Second*0 {
+			minJitter = min(minJitter, absDiff(lastTime, pkt.Rtt))
+			maxJitter = max(minJitter, absDiff(lastTime, pkt.Rtt))
+			sum += absDiff(lastTime, pkt.Rtt)
+		}
+		lastTime = pkt.Rtt
+	}
+	// Runing the pinger
+	if err := pinger.Run(); err != nil {
+		return fmt.Errorf("error running ping : %w", err)
+	}
```

Comment:
This is part of the execution.  I think this will be better in the respective jobs, than being here.

Reply:
If this part is shifted to the jobs package, then the whole function has to be shifted, which doesn't make sense of the `evaluate` package. We can remove the `evaluate` package and do the evaluation in Execute function of the job. Still, then the job package will contain code that doesn't suit the package name, i.e., evaluation of Ping, Jitter, and Monitoring metrics.